monitor/guard.go

```diff
@@ -212,27 +214,42 @@ func (m *Monitor) guardChannel(request *guard.Request) (guarded, delete bool, er
 	}
 
 	// submit guard tx
-	var tx *ethtypes.Transaction
-	switch request.Status {
-	case guard.ChanStatus_Withdrawing:
-		tx, err = m.EthClient.Transactor.Transact(
-			m.guardTxHandler("SnapshotStates", request),
-			func(transactor bind.ContractTransactor, opts *bind.TransactOpts) (*ethtypes.Transaction, error) {
-				return m.EthClient.GetLedger().SnapshotStates(opts, signedSimplexStateArrayBytes)
-			})
-	case guard.ChanStatus_Settling:
-		tx, err = m.EthClient.Transactor.Transact(
-			m.guardTxHandler("IntendSettle", request),
-			func(transactor bind.ContractTransactor, opts *bind.TransactOpts) (*ethtypes.Transaction, error) {
-				return m.EthClient.GetLedger().IntendSettle(opts, signedSimplexStateArrayBytes)
-			})
-	default:
-		return false, false, fmt.Errorf("Invalid guard state %d", request.Status)
-	}
-	if err != nil {
-		return false, false, fmt.Errorf("tx err %w", err)
-	} else {
+	if m.Operator.TxSender == nil {
+		var tx *ethtypes.Transaction
+		switch request.Status {
+		case guard.ChanStatus_Withdrawing:
+			tx, err = m.EthClient.Transactor.Transact(
+				m.guardTxHandler("SnapshotStates", request),
+				func(transactor bind.ContractTransactor, opts *bind.TransactOpts) (*ethtypes.Transaction, error) {
+					return m.EthClient.GetLedger().SnapshotStates(opts, signedSimplexStateArrayBytes)
+				})
+		case guard.ChanStatus_Settling:
+			tx, err = m.EthClient.Transactor.Transact(
+				m.guardTxHandler("IntendSettle", request),
+				func(transactor bind.ContractTransactor, opts *bind.TransactOpts) (*ethtypes.Transaction, error) {
+					return m.EthClient.GetLedger().IntendSettle(opts, signedSimplexStateArrayBytes)
+				})
+		default:
+			return false, false, fmt.Errorf("Invalid guard state %d", request.Status)
+		}
+		if err != nil {
+			return false, false, fmt.Errorf("tx err %w", err)
+		}
 		log.Infof("submitted guard tx %x", tx.Hash())
+	} else {
+		var guardErr error
+		switch request.Status {
+		case guard.ChanStatus_Withdrawing:
+			guardErr = m.Operator.TxSender.SnapshotStates(signedSimplexStateArrayBytes, m.guardJobHandler("SnapshotStates", request))
+		case guard.ChanStatus_Settling:
+			guardErr = m.Operator.TxSender.IntendSettle(signedSimplexStateArrayBytes, m.guardJobHandler("IntendSettle", request))
+		default:
+			return false, false, fmt.Errorf("Invalid guard state %d", request.Status)
+		}
+		if guardErr != nil {
+			return false, false, guardErr
+		}
+		log.Infof("submitted guard tx via TxSender")
 	}
```

Comment:
How about making the `if...else...` block above a separate `sendGuardTx` func?

Reply:
Done.